lib/OptionsDefaulter.js

```diff
@@ -6,7 +6,7 @@
 
 function getProperty(obj, name) {
 	name = name.split(".");
-	for(var i = 0; i < name.length - 1; i++) {
```

Comment:
While `node =< v4.0.0` is the target this needs to keep using `var` as there is a known scope bug for loops in the V8 version `node =< v4.0.0` uses

Reply:
Thanks for reviewing. Should I change let to var here only or everywhere in the for loops?

Reply:
@michael-ciniawsky which bug? Is this snippet affected by this bug?

Reply:
No sry this was a false alarm on my side this bug only seems to affects [`while` loops](https://github.com/nodejs/node/issues/10435)

A ~[5x perf hit](https://github.com/nodejs/node/issues/5815#issuecomment-199473180) it is to be expected , but that was already being mentioned :)

Reply:
You shouldn't use node 4 anyway when caring about performance... ^^